src/internal_events/tcp.rs

```diff
@@ -190,3 +190,32 @@ impl InternalEvent for TcpBytesReceived {
         );
     }
 }
+
+pub struct TcpSocketReceiveError<E> {
+    pub error: E,
+}
+
+impl<E: std::fmt::Display> InternalEvent for TcpSocketReceiveError<E> {
+    fn emit(self) {
+        error!(
+            message = "TCP socket error.",
```

Comment:
```suggestion
            message = "TCP socket receive error.",
```

Mainly because in my sink side socket changes I'm altering the other event to be "TCP socket send error."